contracts/modules/licensing/LicensingModuleV2.sol

```diff
@@ -0,0 +1,360 @@
+            originalIpId,
+            licenseTemplate,
+            licenseTermsId,
+            amount,
+            msg.sender,
+            receiver
+        );
+
+        emit LicenseTokensMinted(
+            msg.sender,
+            originalIpId,
+            licenseTemplate,
+            licenseTermsId,
+            amount,
+            receiver,
+            startLicenseTokenId,
+            endLicenseTokenId
+        );
+    }
+
+    function registerDerivative(
+        address derivativeIpId,
+        address[] calldata originalIpIds,
+        uint256[] calldata licenseTermsIds,
+        address licenseTemplate,
+        bytes calldata royaltyContext
+    ) external nonReentrant verifyPermission(derivativeIpId) {
+        if (originalIpIds.length != licenseTermsIds.length) {
+            revert Errors.LicensingModule__LicenseTermsLengthMismatch(originalIpIds.length, licenseTermsIds.length);
+        }
+        if (originalIpIds.length == 0) {
+            revert Errors.LicensingModule__NoOriginalIp();
+        }
+
+        _verifyIpNotDisputed(derivativeIpId);
+
+        address derivativeIpOwner = IIPAccount(payable(derivativeIpId)).owner();
+        ILicenseTemplate lct = ILicenseTemplate(licenseTemplate);
+        if (!lct.verifyRegisterDerivativeForAll(derivativeIpId, originalIpIds, licenseTermsIds, derivativeIpOwner)) {
+            revert Errors.LicensingModule__LicenseNotCompatibleForDerivative(derivativeIpId);
+        }
+
+        LICENSE_REGISTRY.registerDerivativeIp(derivativeIpId, originalIpIds, licenseTemplate, licenseTermsIds);
+        // pay minting fee
+        (address commonRoyaltyPolicy, bytes[] memory royaltyDatas) = _payMintingFeeForAll(
+            originalIpIds,
+            licenseTermsIds,
+            licenseTemplate,
+            derivativeIpOwner,
+            royaltyContext
+        );
+        // emit event
+        emit DerivativeRegistered(msg.sender, derivativeIpId, originalIpIds, licenseTermsIds, licenseTemplate);
+
+        if (commonRoyaltyPolicy != address(0)) {
+            ROYALTY_MODULE.onLinkToParents(
+                derivativeIpId,
+                commonRoyaltyPolicy,
+                originalIpIds,
+                royaltyDatas,
+                royaltyContext
+            );
+        }
+    }
+
+    function registerDerivativeWithLicenseTokens(
+        address derivativeIpId,
+        uint256[] calldata licenseTokenIds,
+        bytes calldata royaltyContext
+    ) external nonReentrant verifyPermission(derivativeIpId) {
+        if (licenseTokenIds.length == 0) {
+            revert Errors.LicensingModule__NoLicenseToken();
+        }
+
+        address derivativeIpOwner = IIPAccount(payable(derivativeIpId)).owner();
+        (address licenseTemplate, address[] memory originalIpIds, uint256[] memory licenseTermsIds) = LICENSE_NFT
+            .validateLicenseTokensForDerivative(derivativeIpId, derivativeIpOwner, licenseTokenIds);
+
+        _verifyIpNotDisputed(derivativeIpId);
```

Comment:
do you need to do this verification before the line 221?

Reply:
It should be ok. Verifying the license token doesn't require verifying the childIp first. For the positive path, both checks need to pass. For the negative path, the transaction will fail either way.